libs/wingc/src/type_check.rs

```diff
@@ -1542,6 +1542,14 @@ impl<'a> TypeChecker<'a> {
 
 				self.inner_scopes.push(statements);
 			}
+			StmtKind::Break { is_in_loop } => {
+				if !*is_in_loop {
```

Comment:
I think this is redundant if we manage to handle the error reporting in _parser.rs_.

Reply:
We should probably also add an issue for reporting dead code:
```
while true {
  break;
  break; // Should fail on "unreachable code" (out of the scope of this PR)
}
```

Reply:
good callout on the dead code! wonder should it fail or just a warning? (I can create a follow-up issue on it) I'm inclined to the latter one - dead code does not hurt anything but it's "dead"

Reply:
+1 for an issue on dead-code detection.
This is probably only one of a million other cases we can benefit from this.
We currently don't have warnings in wing, we're trying to see how long this approach will hold water..

Reply:
dead-code detection tracking: https://github.com/winglang/wing/issues/1767